integration_tests/route_helpers.go

```diff
@@ -97,3 +105,36 @@ func clearRoutes() {
 	routerDB.C("routes").DropCollection()
 	routerDB.C("backends").DropCollection()
 }
+
+func getMongoSession(uri string) *mgo.Session {
+	const sslSuffix string = "?ssl=true"
```

Comment:
Do we ever specify any other query string parameters within the connection string? This is a very specific mechanism. We could handle it better by parsing the URL ourselves (error handling omitted for brevity):

```go
// Create the DialInfo value, without SSL info
dialInfo, _ := mgo.ParseURL(uri)

// Manually parse we we can work out if it's an SSL connection
url, _ := url.Parse(uri)

// Check for the relevant SSL parameter and add the DialServer func if necessary
query := url.Query()
if values, hasSSL := query["ssl"]; if hasSSL {
    if len(values) > 0 {
        ssl, _ := strconv.ParseBool(values[0])
        if ssl {
            dialInfo.DialServer = func(...) { ... }
        }
    }
}

// Do the Dialling (but see other comment about separating config from call)
session, _ := mgo.DialWithInfo(dialInfo)
return session
```